packages/@react-spectrum/cards/src/CardView.tsx

```diff
@@ -0,0 +1,187 @@
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {CardBase} from './CardBase';
+import {CardViewContext, useCardViewContext} from './CardViewContext';
+import cardViewStyles from './cardview.css';
+import {classNames, useDOMRef, useStyleProps, useUnwrapDOMRef} from '@react-spectrum/utils';
+import {DOMRef, DOMRefValue} from '@react-types/shared';
+import {GridCollection, useGridState} from '@react-stately/grid';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {ProgressCircle} from '@react-spectrum/progress';
+import React, {ReactElement, useMemo, useRef} from 'react';
+import {SpectrumCardViewProps} from '@react-types/cards';
+import {useCollator, useLocale, useMessageFormatter} from '@react-aria/i18n';
+import {useGrid, useGridCell, useGridRow} from '@react-aria/grid';
+import {useListState} from '@react-stately/list';
+import {Virtualizer} from '@react-aria/virtualizer';
+
+function CardView<T extends object>(props: SpectrumCardViewProps<T>, ref: DOMRef<HTMLDivElement>) {
+  let {styleProps} = useStyleProps(props);
+  let domRef = useDOMRef(ref);
+  let {
+    isQuiet,
+    renderEmptyState,
+    layout,
+    loadingState,
+    onLoadMore
+  } = props;
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
+  let cardViewLayout = useMemo(() => typeof layout === 'function' ? new layout({collator}) : layout, [layout, collator]);
+
+  let formatMessage = useMessageFormatter(intlMessages);
+  let {direction} = useLocale();
+  let {collection} = useListState(props);
+
+  let gridCollection = useMemo(() => new GridCollection<T>({
+    columnCount: 1,
+    items: [...collection].map(item => ({
+      // Makes the Grid row use the keys the user provides to the cards so that selection change via interactions returns the card keys
+      type: 'item',
+      key: item.key,
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell',
+        key: `cell-${item.key}`
+      }]
+    }))
+  }), [collection]);
+
+  let state = useGridState({
+    ...props,
+    collection: gridCollection
+  });
+
+  // TODO: need to fix the typescript here, perhaps add a new type in Card types which is a Layout w/ these properties
+  // TODO: double check that this is the correct collection being set (we wanna use the list collection for the keyboard delegate?)
+  // If not, update the gridlayout code to use the gridCollection
```

Comment:
Previously was using the list collection instead of the grid collection since it made some of the layout stuff easier (no need to deal with rows vs cells), but have refactored to use grid collection in a follow up. Will clean up this comment